src/components/reportPage/ReportPage.jsx

```diff
@@ -46,34 +47,80 @@ const ReportPage = (props) => {
     if (files) fetch();
   }, []);
 
+  const handleSearching = (e) => {
+    setSearchString(e.target.value);
+  };
+
   return (
     <>
-      <button className="back-btn" onClick={() => history.goBack()}>
-        &laquo; Go Back
-      </button>
-      <div className="progress-indicator-container">
-        <p>
-          {progressValue} / {files?.length ?? 0} files processed
-        </p>
-        <ProgressBar
-          animated={progressValue != files?.length}
-          className="progress"
-          now={files ? (progressValue / files.length) * 100 + 1 : 0}
-        />
+      <div className="results">
+        <button className="back-btn" onClick={() => history.goBack()}>
+          &laquo; Go Back
+        </button>
+        <div className="progress-indicator-container">
+          <p>
+            {progressValue} / {files?.length ?? 0} files processed
+          </p>
+          <ProgressBar
+            animated={progressValue != files?.length}
+            className="progress"
+            now={files ? (progressValue / files.length) * 100 + 1 : 0}
+          />
+        </div>
+        <div className="search-area">
+          <input
+            type="text"
+            name="search"
+            placeholder="Type name of file here..."
+            className="file-search"
+            value={searchTerm}
+            onChange={handleSearching}
+          />
+        </div>
       </div>
+      {imageResults.length > 0 && <h1 className="results-title">Image Results</h1>}
       {imageResults.length > 0 && <ColourSchemeSelector setColourScheme={setColourScheme} />}
-      <div>
-        {imageResults.length > 0 && <h1 className="results-title">Image Results</h1>}
+      <div className="results">
         <div className="results-container">
-          {imageResults.map((result, index) => (
-            <ImageResult key={`image-${index}`} imageResult={result} colourScheme={colourScheme} />
-          ))}
+          {imageResults.reduce((previousResult, currentResult, index) => {
+            if (
+              searchTerm.length === 0 ||
+              images[index].name.toLowerCase().includes(searchTerm.toLowerCase())
+            ) {
+              return [
+                ...previousResult,
+                <ImageResult
+                  key={`image-${index}`}
+                  imageFile={images[index]}
+                  imageResult={currentResult}
+                  colourScheme={colourScheme}
+                />,
+              ];
+            } else {
+              return [...previousResult];
+            }
+          }, [])}
```

Comment:
Nice search implementation 👍 . I think it would have been easier to just use [filter though](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter)?